src/corecel/sys/ScopedProfiling.cuda.cc

```diff
@@ -37,31 +37,32 @@ nvtxDomainHandle_t domain_handle()
  *
  * Insert it if it doesn't already exist.
  */
-nvtxStringHandle_t message_handle_for(std::string const& message)
+nvtxStringHandle_t message_handle_for(std::string_view message)
 {
     static std::unordered_map<std::string, nvtxStringHandle_t> registry;
     static std::shared_mutex mutex;
 
+    std::string temp_message{message};
```

Comment:
registry can also be `std::unordered_map<std::string_view, nvtxStringHandle_t>` and only instantiate the string when we register the domain.

Reply:
I don't think it can, because some of the input strings are temporaries